cmd/cloudFoundryDeleteService.go

```diff
@@ -30,10 +37,54 @@ func cloudFoundryDeleteService(CloudFoundryDeleteServiceOptions cloudFoundryDele
 	return nil
 }
 
+func cloudFoundryDeleteServiceKeys(CloudFoundryDeleteServiceOptions cloudFoundryDeleteServiceOptions, c execRunner) error {
+
+	var cfFindServiceKeysScript = []string{"service-keys", CloudFoundryDeleteServiceOptions.CfServiceInstance}
+
+	var serviceKeyBytes bytes.Buffer
+	c.Stdout(&serviceKeyBytes)
+
+	err := c.RunExecutable("cf", cfFindServiceKeysScript...)
+	if err != nil {
+		cloudFoundryLogout(c)
+		log.Entry().
+			WithError(err).
+			Fatal("Failed to Delete Service Key, most likely your service doesn't exist")
+	}
+
+	if len(serviceKeyBytes.String()) > 0 {
+		var lines []string = strings.Split(serviceKeyBytes.String(), "\n")
+		if len(lines) <= 4 {
+			log.Entry().Info("No Service Keys active to be deleted")
+			return err
+		}
+		var numberOfServiceKeys = len(lines)
+		log.Entry().WithField("Number of service keys :", numberOfServiceKeys-4).Info("ServiceKey")
+
+		//Deleting all matched Service Keys for Service
+		for i := 3; i <= numberOfServiceKeys-2; i++ {
+			log.Entry().WithField("Deleting Service Key", lines[i]).Info("ServiceKeyDeletion")
+
+			var cfDeleteServiceKeyScript = []string{"delete-service-key", CloudFoundryDeleteServiceOptions.CfServiceInstance, lines[i], "-f"}
+
+			err := c.RunExecutable("cf", cfDeleteServiceKeyScript...)
+			if err != nil {
+				cloudFoundryLogout(c)
+				log.Entry().
+					WithError(err).
+					Fatal("Failed to login to Delete Service Key")
+			}
+		}
```

Comment:
I think this code would become more clear if you declare `var numberOfServiceKeys = len(lines) - 4` and then return an error `if numberOfSericeKeys <= 0`. You would then need to fix the loop of course. Alternatively rename the variable to `numberOfLines`.

Reply:
Tried it before, code doesn't seem to become more clearer for me since you would then need another variable/logic in the for loop. Error is returned anyway if there are equal or less then 4 entries in `numberOfServiceKeys` since the keys will appear from index 4 and higher. What do you think, would it make sense to add a new variable where only the ServiceKeys are stored for clearer code?

Reply:
Like I said, "alternatively rename the variable to `numberOfLines`", then you wouldn't need any other code changes.

Reply:
Ok, changed it accordingly to `numberOfLines`

Reply:
Thanks!